src/dask_histogram/core.py

```diff
@@ -401,27 +401,28 @@ def optimize(
     keys: Hashable | list[Hashable] | set[Hashable],
     **kwargs: Any,
 ) -> Mapping:
-    if not isinstance(keys, (list, set)):
-        keys = [keys]
-    keys = list(flatten(keys))
+    keys = tuple(flatten(keys))
 
     if not isinstance(dsk, HighLevelGraph):
-        dsk = HighLevelGraph.from_collections(id(dsk), dsk, dependencies=())
+        dsk = HighLevelGraph.from_collections(str(id(dsk)), dsk, dependencies=())
 
+    dsk = optimize_blockwise(dsk, keys=keys)
+    dsk = fuse_roots(dsk, keys=keys)  # type: ignore
+    dsk = dsk.cull(set(keys))  # type: ignore
+    return dsk
+
+
+def _get_optimization_function():
     # Here we try to run optimizations from dask-awkward (if we detect
     # that dask-awkward has been imported). There is no cost to
     # running this optimization even in cases where it's unncessary
-    # because if no AwkwardInputLayers from daks-awkward are not
+    # because if no AwkwardInputLayers from dask-awkward are
     # detected then the original graph is returned unchanged.
     if dask.config.get("awkward", default=False):
```

Comment:
This is populated when dak is imported, correct? This import has to have happened if any dak components are in the graph.

Reply:
Yes - so the only way this fails is if you're loading a pickled task-graph with dask-awkward stuff in it... But even then I think that requires you've imported the module? So I think this is rather safe.